pkg/commands/git_commands/commit_loader.go

```diff
@@ -364,23 +358,40 @@ func (self *CommitLoader) setCommitMergedStatuses(refName string, commits []*mod
 			commits[i].Status = models.StatusMerged
 		}
 	}
-	return commits, nil
+	return commits
 }
 
-func (self *CommitLoader) getMergeBase(refName string) (string, error) {
-	info, err := self.getCurrentBranchInfo()
-	if err != nil {
-		return "", err
-	}
+func (self *CommitLoader) getMergeBase(refName string) string {
+	quotedBaseBranches := strings.Join(
+		lo.FilterMap(self.UserConfig.Git.BaseBranches, func(branch string, _ int) (string, bool) {
+			// We need to filter out base branches that don't exist, as they
+			// would make the git merge-base call fail with an error
+			if !self.branchExists(branch) {
+				return "", false
+			}
 
-	baseBranch := "master"
-	if strings.HasPrefix(info.RefName, "feature/") {
-		baseBranch = "develop"
+			return self.cmd.Quote(branch), true
+		}), " ")
+
+	if quotedBaseBranches == "" {
+		return ""
 	}
 
 	// swallowing error because it's not a big deal; probably because there are no commits yet
-	output, _ := self.cmd.New(fmt.Sprintf("git merge-base %s %s", self.cmd.Quote(refName), self.cmd.Quote(baseBranch))).DontLog().RunWithOutput()
-	return ignoringWarnings(output), nil
+	output, _ := self.cmd.New(fmt.Sprintf("git merge-base %s %s", self.cmd.Quote(refName), quotedBaseBranches)).DontLog().RunWithOutput()
```

Comment:
I'd add a comment explaining the behaviour that we get with this approach i.e. whichever branch is closest to the current branch will be used to get the merge base

Reply:
I added one in the new version.